### PR TITLE
Make S3 dependencies optional

### PR DESCRIPTION
**Pull request by floscha**
Based on the discussion in #340, this PR makes all S3-related dependencies optional, considering that S3-usage is presumably limited to a smaller range of users. However, the dependencies were moved to *test-requirements.txt* so that unit tests still run.

One corner case I left out is [S3 buckets](https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingBucket.html#access-bucket-intro) which are not supported right now. Adding such support would (IMHO) require adding some regular expressions which would unnecessarily clutter the code. If preferred, I can add those nonetheless.

### CODE REVIEW

File: featuretools/primitives/install.py [hunk 1]
```diff
@@ -100,12 +98,20 @@ def download_archive(uri):
     local_archive = os.path.join(get_installation_temp_dir(), filename)
 
     with open(local_archive, 'wb') as f:
-        try:
+        if uri.startswith('s3://') or uri.startswith('https://s3'):
+            try:
+                import s3fs
+                from botocore.exceptions import NoCredentialsError
+            except ImportError:
+                raise ImportError("The s3fs library is required to handle s3 files")
```
Comment (kmax12) [inline featuretools/primitives/install.py]: s3fs is actually only used as a fallback since `smart_open` cannot handle the anonymous clients which leads to the `NoCredentialsError` case. smart open can open s3 files in other cases.

see this issue: https://github.com/RaRe-Technologies/smart_open/issues/250

perhaps move this try/except into the `NoCredentialsError` block which is the only place s3fs is used.

**Comment by kmax12** [comment]
do we need to add this check to the `load_retail` or `load_flight` functions as well?

**Comment by floscha** [comment]
Both `load_retail` and `load_flight` work without having `s3fs` installed. However, I noticed that `smart_open` crashes when `boto3` isn't installed (which already happens when calling `import featuretools`), so I added it back to *requirements.txt* with 401f4d0.

File: featuretools/primitives/install.py (hunk 1 above)
Comment (floscha) [reply featuretools/primitives/install.py]: I see. I was assuming that `smart_open` also uses `s3fs` under the hood. I've now placed the `s3fs` import as you suggested.

**Comment by kmax12** [comment]
Looks good merging
